include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_for.h

```diff
@@ -127,46 +151,42 @@ struct __parallel_for_large_submitter<__internal::__optional_kernel_name<_Name..
         const std::size_t __work_group_size =
             oneapi::dpl::__internal::__max_work_group_size(__q, __max_work_group_size);
         const std::uint32_t __max_cu = oneapi::dpl::__internal::__max_compute_units(__q);
-        return __work_group_size * _Fp::__preferred_iters_per_item * __max_cu;
+        return __work_group_size * __params_t::__iters_per_item * __max_cu;
     }
 
     template <typename _Fp, typename _Index, typename... _Ranges>
     __future<sycl::event>
     operator()(sycl::queue& __q, _Fp __brick, _Index __count, _Ranges&&... __rngs) const
     {
+        using __params_t = __pfor_params<true /*__enable_tuning*/, _Fp, _Ranges...>;
         assert(oneapi::dpl::__ranges::__get_first_range_size(__rngs...) > 0);
         const std::size_t __work_group_size =
             oneapi::dpl::__internal::__max_work_group_size(__q, __max_work_group_size);
         _PRINT_INFO_IN_DEBUG_MODE(__q);
         auto __event = __q.submit([__rngs..., __brick, __work_group_size, __count](sycl::handler& __cgh) {
             //get an access to data under SYCL buffer:
             oneapi::dpl::__ranges::__require_access(__cgh, __rngs...);
-            constexpr std::uint8_t __iters_per_work_item = _Fp::__preferred_iters_per_item;
-            constexpr std::uint8_t __vector_size = _Fp::__preferred_vector_size;
+            constexpr std::uint8_t __iters_per_work_item = __params_t::__iters_per_item;
+            constexpr std::uint8_t __vector_size = __params_t::__vector_size;
             const std::size_t __num_groups = oneapi::dpl::__internal::__dpl_ceiling_div(
                 __count, (__work_group_size * __vector_size * __iters_per_work_item));
             __cgh.parallel_for<_Name...>(
                 sycl::nd_range(sycl::range<1>(__num_groups * __work_group_size), sycl::range<1>(__work_group_size)),
                 [=](sycl::nd_item</*dim=*/1> __item) {
+                    __params_t __params;
                     const auto [__idx, __stride, __is_full] =
                         __stride_recommender(__item, __count, __iters_per_work_item, __vector_size, __work_group_size);
                     __strided_loop<__iters_per_work_item> __execute_loop{static_cast<std::size_t>(__count)};
-                    if (__is_full)
-                    {
-                        __execute_loop(std::true_type{}, __idx, __stride, __brick, __rngs...);
-                    }
-                    // If we are not full, then take this branch only if there is work to process.
-                    else if (__idx < __count)
-                    {
-                        __execute_loop(std::false_type{}, __idx, __stride, __brick, __rngs...);
-                    }
+                    __is_full ? __execute_loop(std::true_type{}, __idx, __stride, __brick, __params, __rngs...)
+                              : __execute_loop(std::false_type{}, __idx, __stride, __brick, __params, __rngs...);
```

Comment:
Is it measurably faster to use a ternary operator here or something?
I find it less readable personally to write it this way rather than `if`, `else`.

Reply:
It is not faster, I had just adopted the suggestion in https://github.com/uxlfoundation/oneDPL/pull/2165#discussion_r2065811416.

`if...else` is probably more conventional than a ternary with functions that have no return values, so I can switch back.